src/client/components/aea-theme/aea-menu2.ls

```diff
@@ -0,0 +1,82 @@
+component-name = "aea-menu"
+Ractive.components[component-name] = Ractive.extend do
+    template: RACTIVE_PREPARSE('aea-menu2.pug')
+    isolated: yes
+    data: ->
+        is-in-debug-mode: no
+        is-menu-open: no
+    onrender: ->
+        __ = @
+        main-sidebar = $ @find \.main-sidebar
+        main-sidebar-button = $ @find \.main-sidebar-button
+
+        if $ window .width! > 1200
+            __.set \isMenuOpen, yes
+            main-sidebar .removeClass \collapsed
+
+        update-debug-mode = ->
+            if __.get \isInDebugMode
+                $ @find \.main-sidebar .addClass \debug
+            else
+                $ @find \.main-sidebar .removeClass \debug
+
+        @observe \isInDebugMode, update-debug-mode
+
+        @observe \isMenuOpen, ->
+            if __.get \isMenuOpen
```

Comment:
Since this callback will be fired *after* the variable's value have been changed, `isMenuOpen == true` may be interpreted as **the user wants to open the menu** - so the variable won't reflects the old state. Hence the menu opening procedures whilst menu open (`isMenuOpen == true`).